### PR TITLE
Fixed issue #3724

### LINKED ISSUE #3724: `pybamm.Experiment.read_termination`, arg `termination` not documented or given a type. The return value is also not documented

### PR DESCRIPTION
**Pull request by jonchapman1**
Documentation added for pybamm.Experiment.read_termination

# Description

Testing the workflow for participants of the developed training workshop in April 2024 to get used to contributing with simple fixes in advance of the workshop. I added documentation for pybamm.Experiment.read_termination

Fixes #3724

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [X] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [X] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [X] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by tinosulzer — APPROVED** [review]

File: pybamm/experiment/experiment.py [hunk 1]
```diff
@@ -141,6 +141,21 @@ def __repr__(self):
     def read_termination(self, termination):
         """
         Read the termination reason. If this condition is hit, the experiment will stop.
+
+        Parameters
+        ----------
+        termination : list[str], optional
+           A single string, or a list of strings, representing the conditions to terminate the experiment.
+           Only capacity or voltage can be provided as a termination reason.
+           e.g. '4 Ah capacity' or ['80% capacity', '2.5 V']
```
Comment (kratman) [inline pybamm/experiment/experiment.py:150]: If we are going to say that the type is a ```list[str]``` then we should enforce that as well. The signature can be changed to ```def read_termination(self, termination: list[str]):``` and the check for the string being a single string can be removed.

**Review by kratman — CHANGES_REQUESTED** [review]

File: pybamm/experiment/experiment.py (hunk 1 above)
Comment (brosaplanella) [reply pybamm/experiment/experiment.py:150]: I think at the moment we accept both a string and a list, which is probably not ideal but there quite a few other methods work like this and changing them at the moment is a big change.

For context, this PR from Jon was to test the pre-workshop exercise instructions we are giving attendees of our next Hackathon. To ensure they know how to use Github we ask them to add a docstring somewhere and open a PR (this way we improve documentation too).

I think probably the best solution for the moment would be to write
```
termination : str or list[str], optional
```
which reflects current use.

Comment (kratman) [reply pybamm/experiment/experiment.py:150]: If this was only to update documentation that is fine, but I would say we should open a ticket for making it just a ```list[str]```. Long term it reduces code complexity to support only a single type of input for interfaces.

Comment (kratman) [reply pybamm/experiment/experiment.py:150]: It is always good to clean up as we go, even for easy PRs

**Review by brosaplanella — APPROVED** [review]
Looks good to me, thanks Jon!

**Review by kratman — APPROVED** [review]
Removing my requested changes after discussion with @brosaplanella

I do think we should make a ticket to only support ```list[str]``` for the type though. It makes interfaces much nicer when there is only 1 input type and requires less error handling. For users this just requires putting [] around their single strings, so it is not a huge change.

**Comment by rtimms** [comment]
@all-contributors please add @jonchapman1 for docs

**Pull request merged**